### ISSUE TITLE
Investigate an incosistency issue reported by test-harness

### ISSUE THREAD

**Issue by ggoranov** [dusk-network/dusk-blockchain#697]
The inconsistency is noticed on 60-hours (test-harness) system test. This might be a regression from PR 690.

Attached is a goroutines backtrace of the problematic node:
 
[deadlock-10-Oct.txt](https://github.com/dusk-network/dusk-blockchain/files/5363761/deadlock-10-Oct.txt)

```
NODE:9500>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45 +0200 GMT"}}]}}
NODE:9501>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9502>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9503>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9504>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9505>>
{"data":{"blocks":[{"header":{"hash":"58a11969b339906354e42e361ed91f84bd4dd3c42d443ec1ac1b0d1ef0d0cd89","height":52796,"timestamp":"2020-10-10 16:18:03  +0200 GMT"}}]}}
NODE:9506>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9507>>
{"data":{"blocks":[{"header":{"hash":"5f587ee34fb63ac6d0435681a04df9621dbb0c3d95c9d5ac38c8bfb9459d8e02","height":92589,"timestamp":"2020-10-11 08:36:49  +0200 GMT"}}]}}
NODE:9508>>
{"data":{"blocks":[{"header":{"hash":"dea745e2c2a965cfbc0f87eeb9dc23ad212406a1a2a33197d4f08e44f7cf346a","height":155521,"timestamp":"2020-10-12 10:44:45  +0200 GMT"}}]}}
NODE:9509>>
{"data":{"blocks":[{"header":{"hash":"1a0603c040859719f628689e61ed36ef6f53bcc3e83de23dadfcc2f68189d40a","height":155520,"timestamp":"2020-10-12 10:44:43  +0200 GMT"}}]}}
```

**Comment by autholykos** [comment]
Closed via #706

**Comment by autholykos** [comment]
Apparently this might be the cause of an unexpected stall happened after mergin #706
